pkg/fake_gcloud/lib/mem_datastore.dart

```diff
@@ -218,6 +219,101 @@ class MemDatastore implements Datastore {
   Future<void> rollback(Transaction transaction) async {
     return null;
   }
+
+  /// Serializes the content of the Datastore to the [sink], with a line-by-line
+  /// JSON-encoded data format.
+  void writeTo(StringSink sink) async {
+    _entities.forEach((_, entity) {
+      sink.writeln(json.encode({'entity': _encodeEntity(entity)}));
+    });
+    sink.writeln(json.encode({'_unusedId': _unusedId}));
+  }
+
+  /// Reads content as a line-by-line JSON-encoded data format.
+  void readFrom(Iterable<String> lines) {
```

Comment:
Wouldn't it be better to make this an async static function that returns an instance of `MemDatastore`?
Same with `MemStorage` ?

Reply:
I would also suggest taking `Stream<List<int>>` as input, or `List<int>`

Reply:
Just nits...

Reply:
Note, one could even argue that we should expose `MemDatastore` as a type and just export a function that returns an instance of `Datastore` and have `_MemDatastore` be an internal type..

For saving we could just have a function which takes a `Datastore` instances, casts to `_MemDatastore` or throws an `ArgumentError`.

Reply:
I think hiding the type doesn't provide much value, because in the test we should know about it at the time of setup, and also when (re)storing the state. As long as the package is imported as dev_dependency only, we are in the clear. Maybe we could detect in their constructor if we are running inside appengine, and if we are, throw an error - making sure we never use it in prod?

Reply:
Re: `List<int>` vs. Strings... I think the current API is useful for in-memory tests and to plug into using the `File` methods. I'll reconsider when implementing the actual file handling for local development.